client/web/src/insights/pages/search-insight-creation/components/form-series-input/FormSeriesInput.tsx

```diff
@@ -0,0 +1,182 @@
+    useImperativeHandle(innerRef, () => ({
+        focus: () => {
+            if (nameField.meta.error) {
+                return nameReference.current?.focus()
+            }
+
+            if (queryField.meta.error) {
+                return queryReference.current?.focus()
+            }
+
+            nameReference.current?.focus()
+        },
+    }))
+
+    useEffect(() => {
+        if (autofocus) {
+            nameReference.current?.focus()
+        }
+    }, [autofocus])
+
+    const handleSubmitButton = useCallback(
+        async (event: React.MouseEvent) => {
+            // handleSubmit work with form element and use form event
+            // but we can't have sub forms for the sake of semantics.
+            // if this case synthetic event of totally comparable with event
+            // from submit button.
+            await handleSubmit((event as unknown) as React.SyntheticEvent<HTMLFormElement>)
+
+            if (nameField.meta.error) {
+                event.preventDefault()
+                return nameReference.current?.focus()
+            }
+
+            if (queryField.meta.error) {
+                event.preventDefault()
+                return queryReference.current?.focus()
+            }
+        },
+        [handleSubmit, nameField.meta.error, queryField.meta.error]
+    )
+
+    return (
+        <div className={classnames('d-flex flex-column', className)}>
+            <InputField
+                title="Name"
+                placeholder="Example: Function component"
+                description="Name shown in the legend and tooltip"
+                valid={(hasNameControlledValue || nameField.meta.touched) && nameField.meta.valid}
+                error={nameField.meta.touched && nameField.meta.error}
+                {...nameField.input}
+                ref={nameReference}
+            />
+
+            <InputField
+                title="Query"
+                placeholder="Example: patternType:regexp const\\s\\w+:\\s(React\\.)?FunctionComponent"
+                description={
+                    <span>
+                        Do not include the <code>repo:</code> filter as it will be added automatically for the current
+                        repository
+                    </span>
+                }
+                valid={(hasQueryControlledValue || queryField.meta.touched) && queryField.meta.valid}
+                error={queryField.meta.touched && queryField.meta.error}
+                className="mt-4"
+                {...queryField.input}
+                ref={queryReference}
+            />
+
+            <FormColorInput
+                name="series color group"
+                title="Color"
+                className="mt-4"
+                value={colorField.input.value}
+                onChange={colorField.input.onChange}
+            />
+
+            <div className="mt-4">
+                <button type="submit" onClick={handleSubmitButton} className="btn btn-light">
```

Comment:
This should be `btn-secondary` according to the design (`btn-light` we don't really use, it has very bad contrast and doesn't work well with light/dark theme)

```suggestion
                <button type="submit" onClick={handleSubmitButton} className="btn btn-secondary">
```